docs/upgrade-guide-2.md

```diff
@@ -0,0 +1,369 @@
+---
+title: Upgrade guide to OpenAPI.NET 2.1 
+description: Learn how to upgrade your OpenAPI.NET version from 1.6 to 2.0 
+author: rachit.malik
+ms.author: malikrachit
+ms.topic: conceptual
+---
+# Introduction
+
+We are excited to announce the preview of a new version of the OpenAPI.NET library!  
+OpenAPI.NET v2 is a major update to the OpenAPI.NET library. This release includes a number of performance improvements, API enhancements, and support for OpenAPI v3.1.
+
+## The biggest update ever
+
+Since the release of the first version of the OpenAPI.NET library in 2018, there has not been a major version update to the library. With the addition of support for OpenAPI v3.1 it was necessary to make some breaking changes. With this opportunity, we have taken the time to make some other improvements to the library, based on the experience we have gained supporting a large community of users for the last six years .
+
+# Performance Improvements
+
+One of the key features of OpenAPI.NET is its performance. This version makes it possible to parse JSON based OpenAPI descriptions even faster. OpenAPI.NET v1 relied on the excellent YamlSharp library for parsing both JSON and YAML files. With OpenAPI.NET v2 we are relying on System.Text.Json for parsing JSON files. For YAML files, we continue to use YamlSharp to parse YAML but then convert to JsonNodes for processing. This allows us to take advantage of the performance improvements in System.Text.Json while still supporting YAML files.
+
+In v1, instances of `$ref` were resolved in a second pass of the document to ensure the target of the reference has been parsed before attempting to resolve it. In v2, reference targets are lazily resolved when reference objects are accessed. This improves load time performance for documents that make heavy use of references.
+
+[How does this change the behaviour of external references?]
+
+## Reduced Dependencies
+
+In OpenAPI v1, it was necessary to include the Microsoft.OpenApi.Readers library to be able to read OpenAPI descriptions in either YAML or JSON.  In OpenAPI.NET v2, the core Microsoft.OpenAPI library can both read and write JSON.  It is only necessary to use the newly renamed [Microsoft.OpenApi.YamlReader](https://www.nuget.org/packages/Microsoft.OpenApi.YamlReader/) library if you need YAML support. This allows teams who are only working in JSON to avoid the additional dependency and therefore eliminate all non-.NET library references.
```

Comment:
A code snippet of how to do this would be useful. Presumably:

```csharp
var settings = new OpenApiReaderSettings();
settings.AddYamlReader();

var result = OpenApiDocument.LoadAsync(openApiString, settings: settings);
```

Reply:
fixed, thanks